demos/python_demos/multi_camera_multi_target_tracking/utils/video.py

```diff
@@ -20,55 +20,36 @@ def __init__(self, sources):
         assert sources
         self.captures = []
         self.transforms = []
-
-        try:
-            sources = [int(src) for src in sources]
-            mode = 'cam'
-        except ValueError:
-            mode = 'video'
-
-        if mode == 'cam':
-            for id in sources:
-                log.info('Connection  cam {}'.format(id))
-                cap = cv.VideoCapture(id)
-                cap.set(cv.CAP_PROP_FRAME_WIDTH, 1280)
-                cap.set(cv.CAP_PROP_FRAME_HEIGHT, 720)
-                cap.set(cv.CAP_PROP_FPS, 30)
-                cap.set(cv.CAP_PROP_FOURCC, cv.VideoWriter_fourcc(*'MJPG'))
-                assert cap.isOpened()
-                self.captures.append(cap)
-        else:
-            for video_path in sources:
-                log.info('Opening file {}'.format(video_path))
-                cap = cv.VideoCapture(video_path)
-                assert cap.isOpened()
-                self.captures.append(cap)
+        self.frame_size = []
+        self.fps = []
+        for src in sources:
+            cap = open_images_capture(src, False)
+            fps = cap.fps()
+            self.captures.append(cap)
+            self.fps.append(fps)
 
     def add_transform(self, t):
         self.transforms.append(t)
 
     def get_frames(self):
         frames = []
         for capture in self.captures:
-            has_frame, frame = capture.read()
-            if has_frame:
-                for t in self.transforms:
-                    frame = t(frame)
-                frames.append(frame)
+            frame = capture.read()
+            if frame is None:
+                raise RuntimeError("Can't read an image from the input")
+            frame_size = frame.shape
+            self.frame_size.append((frame_size[1], frame_size[0]))
```

Comment:
This will end up storing the whole history of frame sizes. Another problem is `get_frames()` is executed in a different thread so `self.frame_size` may be empty when `get_source_parameters()` is called. I suggest to `open()` `VideoWriter` after the first successful call of `frames = thread_body.frames_queue.get_nowait()` and extract sizes from `frames`.